bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Component.java

```diff
@@ -55,12 +52,10 @@ default String getId() {
      * Returns the data layer information associated with the component
      *
      * @return {@link ComponentData} object associated with the component
-     *
      * @since com.adobe.cq.wcm.core.components.models 12.12.0
      */
     @Nullable
     @JsonProperty("dataLayer")
-    @JsonSerialize(using = ComponentDataModelSerializer.class)
```

Comment:
The removal of this line breaks the serialization of the data model. 
The serialized format should be along the lines of:
```JSON
{
    "<ComponentID>": {
        "....serialized ComponentData..."
    }
}
```

The results of this change cause 
```JSON
{
    "....serialized ComponentData..."
}
```

Reply:
Here's a diff of actual test output (expected on left, actual on right).
Please make sure to run the tests.

![Screenshot from 2021-02-06 03-27-23](https://user-images.githubusercontent.com/19987287/107112097-54c4d400-682b-11eb-9950-3b72f462dce9.png)

Reply:
> This PR should probably be made against `adobe:development` and not `adobe:master`.

Thanks, I changed the base of this PR.

> The removal of this line breaks the serialization of the data model.
Yes, it did. The change was intentional, though. As I mentioned in the PR's description it's probably not a good idea to reference the serializer (which is located inside "internal") from within the public API.

As far as I can tell the serializer is only necessary to be used inside Unit tests, correct? Are consumers of the API expecting to be able to serialize instances of `Component` directly which in turn should use the Serializer? Or are consumers only supposed to call `ComponentData#getJson`?

(This might be a concern especially when using `ComponentExporter`)

Reply:
`Component/ContainerExporter` serialization is important for SPA developers (which consume `.model.json` output instead of rendered HTML attributes to feed the DataLayer).